pkg/asset/installconfig/aws/metadata.go

```diff
@@ -134,3 +138,25 @@ func (m *Metadata) VPC(ctx context.Context) (string, error) {
 
 	return m.vpc, nil
 }
+
+// IgnitionPresignedURL returns an S3 bucket name and a presigned URL for it
+func (m *Metadata) IgnitionPresignedURL(ctx context.Context, infraID string) (string, string, error) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	session, err := m.unlockedSession(ctx)
+	if err != nil {
+		return "", "", err
+	}
+
+	bucketName := fmt.Sprintf("%s-bootstrap", infraID)
+
+	client := s3.New(session, aws.NewConfig().WithRegion(m.Region))
+	req, _ := client.GetObjectRequest(&s3.GetObjectInput{
+		Bucket: aws.String(bucketName),
+		Key:    aws.String("bootstrap.ign"),
```

Comment:
Is this key a secret? If so, should we use something less static?

Reply:
No, it's just the name of the bucket. It's necessary to get a presigned URL.